### ISSUE TITLE
Consumer - Key Auth, Only works on first api configuration

### ISSUE THREAD

**Issue by fabianrdz** [Kong/kong#3299]
I've been trying kong since a few days ago and i've been facing this issue for a while.

Following the docs
I added an api.
https://getkong.org/docs/0.12.x/getting-started/adding-your-api/

Then i added the key auth pluggin.
https://getkong.org/docs/0.12.x/getting-started/enabling-plugins/

And then i added a consumer
https://getkong.org/docs/0.12.x/getting-started/adding-consumers/

This worked fine.

When i add a new api, the plugin and a new consumer for that new api, it just doesn't work, it allways get back with a 401 Unauthorized.
If i disable key-auth for that new api, it works fine.

After that, i delete the first example api, plugin, and consumer, and key-auth didn't work again with any other apy.

I have to reset the database, and configure a new api, and again, it just works on first configuration.

kong version 0.12.3
postgres 9.4 (docker)

**Comment by bungle** [comment]
Hi,

Do you have a commands that you did? A minimal example where this happens. In a mean time I will try this by following your description.

**Comment by fabianrdz** [comment]
I executed the same commands as you.

curl -i -X POST --url http://localhost:8001/apis/  --data 'name=example-api'   --data 'hosts=example.com'   --data 'upstream_url=http://mockbin.org'
curl -i -X POST  --url http://localhost:8001/apis/example-api/plugins/  --data 'name=key-auth'
curl -i -X POST   --url http://localhost:8001/consumers/    --data "username=Jason" 
curl -i -X POST   --url http://localhost:8001/consumers/Jason/key-auth/   --data 'key=ENTER_KEY_HERE'
curl -i -X GET    --url http://localhost:8000    --header "Host: example.com"    --header "apikey: ENTER_KEY_HERE"


curl -i -X POST   --url http://localhost:8001/apis/    --data 'name=example2-api'    --data 'hosts=example2.com'    --data 'upstream_url=http://mockbin.org'
curl -i -X POST   --url http://localhost:8001/apis/example2-api/plugins/   --data 'name=key-auth'
curl -i -X POST   --url http://localhost:8001/consumers/   --data "username=Jason2" 
curl -i -X POST    --url http://localhost:8001/consumers/Jason2/key-auth/   --data 'key=ENTER_KEY_HERE2'
curl -i -X GET    --url http://localhost:8000    --header "Host: example2.com"    --header "apikey: ENTER_KEY_HERE2"

This worked normally, but after a while i executed these

https://user-images.githubusercontent.com/16723566/37487463-38fe31c4-2857-11e8-93e5-54d02f2b8c3c.png

Basically the same, but different names.

I'm not sure if i'm doing something that corrupts the database, because after this happen for the first time, it doesn't recover, and the only solution(for me) is reset my db

**Comment by bungle** [comment]
Can you make that `png` available as text, so that I can execute it similarly on my side. You never know if there are weird unicode sequences or something. Do you get anything in Kong `error.log`?

**Comment by bungle** [comment]
Hi,

I tried your example, and it worked as expected.

Now I see that you created an API that points to another API on Kong, but on port `8334`? Is my assumption correct?

```
http://localhost:8443/cas/login
```

Is this also served by same Kong and protected with key-auth? Can you try to configure Nginx to use just 1 worker process `nginx_worker_processes` config parameter or with ENV var `KONG_NGINX_WORKER_PROCESSES=1`?

**Comment by fabianrdz** [comment]
Hi,

In 8443 is runnig other application that i'm wonrking on, thats the endpoint that i want to reach through kong.

I will tray the nginx configuration and see the results.

Thank you!

These are the previous configurations
aaaa@aaaa:~$ curl -i -X POST   --url http://localhost:8001/apis/    --data 'name=final-api'    --data 'hosts=final.api.com'    --data 'upstream_url=https://localhost:8443/cas/login'
HTTP/1.1 201 Created
Date: Thu, 15 Mar 2018 19:40:58 GMT
Content-Type: application/json; charset=utf-8
Transfer-Encoding: chunked
Connection: keep-alive
Access-Control-Allow-Origin: *
Server: kong/0.12.3

{"created_at":1521142858691,"strip_uri":true,"id":"ea9d0d23-9914-4a7a-9441-81e4fe880ff6","hosts":["final.api.com"],"name":"final-api","http_if_terminated":false,"preserve_host":false,"upstream_url":"https:\/\/localhost:8443\/cas\/login","upstream_connect_timeout":60000,"upstream_send_timeout":60000,"upstream_read_timeout":60000,"retries":5,"https_only":false}
aaaa@aaaa:~$ curl -i -X POST   --url http://localhost:8001/apis/final-api/plugins/   --data 'name=key-auth'
HTTP/1.1 201 Created
Date: Thu, 15 Mar 2018 19:41:02 GMT
Content-Type: application/json; charset=utf-8
Transfer-Encoding: chunked
Connection: keep-alive
Access-Control-Allow-Origin: *
Server: kong/0.12.3

{"created_at":1521142862000,"config":{"key_in_body":false,"run_on_preflight":true,"anonymous":"","hide_credentials":false,"key_names":["function"]},"id":"5cec6e96-02d9-421e-9cf3-fd40c3beafff","name":"key-auth","api_id":"ea9d0d23-9914-4a7a-9441-81e4fe880ff6","enabled":true}
aaaa@aaaa:~$ curl -i -X POST   --url http://localhost:8001/consumers/   --data "username=final" 
HTTP/1.1 201 Created
Date: Thu, 15 Mar 2018 19:41:15 GMT
Content-Type: application/json; charset=utf-8
Transfer-Encoding: chunked
Connection: keep-alive
Access-Control-Allow-Origin: *
Server: kong/0.12.3

{"created_at":1521142876000,"username":"final","id":"3eb288af-4f5f-438d-9d51-a0b227321b65"}
aaaa@aaaa:~$ curl -i -X POST    --url http://localhost:8001/consumers/final/key-auth/   --data 'key=finalkey'
HTTP/1.1 201 Created
Date: Thu, 15 Mar 2018 19:41:19 GMT
Content-Type: application/json; charset=utf-8
Transfer-Encoding: chunked
Connection: keep-alive
Access-Control-Allow-Origin: *
Server: kong/0.12.3

{"id":"9d7be6cf-3e07-49d3-b139-055a7de0df79","created_at":1521142879000,"key":"finalkey","consumer_id":"3eb288af-4f5f-438d-9d51-a0b227321b65"}
aaaa@aaaa:~$ curl -i -X GET    --url http://localhost:8000    --header "Host: final.api.com"    --header "apikey: finalkey"
HTTP/1.1 401 Unauthorized
Date: Thu, 15 Mar 2018 19:41:22 GMT
Content-Type: application/json; charset=utf-8
Transfer-Encoding: chunked
Connection: keep-alive
WWW-Authenticate: Key realm="kong"
Server: kong/0.12.3

{"message":"No API key found in request"}

**Comment by bungle** [comment]
I have a feeling that this got fixed with:
https://github.com/Kong/kong/pull/3348

**Comment by bungle** [comment]
Please reopen if that was not a case! Thank you.

**Comment by fabianrdz** [comment]
Thanks!
